src/extension.rs

```diff
@@ -82,6 +105,16 @@ impl Extension {
         }
     }
 
+    /// Create a new extension out of a &str
+    pub fn from_str(ecode: i32, edata: &str) -> Self {
```

Comment:
Does this not take a NiftiECode as an argument for some reason?

Reply:
It's fine for now.  You _could_ simply make these `const` values in the crate and leave it to the end user to do validation.  That's how I handle the mini header in our code.